### PR TITLE
Add adapter module for the foldl package to conduit-extra

### LINKED ISSUE #311: Integration with the foldl library
This is only a question not an actual issue:

At some point `conduit-extra` contained the necessary boilerplate to integrate with the `foldl` library (which I personally find quite pleasant to use). However, that code was removed in https://github.com/snoyberg/conduit/commit/147475db76b7a70209b0f6acaa056e82cbed6b64#diff-ba10cb6fed8c36fb57edf7306612a142 without any explanation. Is there a better way to achieve this these days or does the functionality still exist in some other form? If not, are you open to adding it again (probably in the form used by e.g. pipes that does not incur a dependency on the `foldl` package).

### PR DESCRIPTION
**Pull request by cocreature**
fixes #311

### CODE REVIEW

File: conduit-extra/Data/Conduit/Foldl.hs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{-# LANGUAGE RankNTypes #-}
+-- | Adapter module to work with the <https://hackage.haskell.org/package/foldl foldl> package.
```
Comment (snoyberg) [inline conduit-extra/Data/Conduit/Foldl.hs]: Can you add a `@since` comment here and to the functions?

File: conduit-extra/conduit-extra.cabal [hunk 2]
```diff
@@ -24,6 +24,7 @@ Library
                        Data.Conduit.Blaze
                        Data.Conduit.ByteString.Builder
                        Data.Conduit.Filesystem
+                       Data.Conduit.Foldl
```
Comment (snoyberg) [inline conduit-extra/conduit-extra.cabal]: Can you also do a version bump and update the ChangeLog?

Comment (cocreature) [inline conduit-extra/conduit-extra.cabal]: done

File: conduit-extra/Data/Conduit/Foldl.hs (hunk 1 above)
Comment (cocreature) [inline conduit-extra/Data/Conduit/Foldl.hs]: done

**Comment by snoyberg** [comment]
Looks like you're working off of an old commit, there are merge conflicts.

**Comment by cocreature** [comment]
Ahrg, sorry about that. rebased.

**Comment by snoyberg** [comment]
Awesome. I'm just going to wait for Travis to go green and then merge.

**Pull request merged**